vm/ubpf_jit_x86_64.c

```diff
@@ -491,17 +492,26 @@ translate(struct ubpf_vm *vm, struct jit_state *state, char **errmsg)
     emit1(state, 0xc3); /* ret */
 
     /* Division by zero handler */
+
+    // Save the address of the start of the divide by zero handler.
     state->div_by_zero_loc = state->offset;
-    const char *div_by_zero_fmt = "uBPF error: division by zero at PC %u\n";
+
     // RCX is the first parameter register for Windows, so first save the value.
     emit_mov(state, RCX, platform_parameter_registers[2]); /* muldivmod stored pc in RCX */
+    emit_string_load(state, platform_parameter_registers[1], UBPF_STRING_ID_DIVIDE_BY_ZERO);
     emit_load_imm(state, platform_parameter_registers[0], (uintptr_t)stderr);
-    emit_load_imm(state, platform_parameter_registers[1], (uintptr_t)div_by_zero_fmt);
     emit_call(state, vm->error_printf);
 
     emit_load_imm(state, map_register(0), -1);
     emit_jmp(state, TARGET_PC_EXIT);
 
+
+    // Emit string table.
```

Comment:
Is it possible to mark the string table part as non-exec. I guess the answer is no. But if this could be somehow in a non-executable "section" of the JIT-ed code that would be safer IMO.

Reply:
Not easily. It would require returning data segment and code segment separately and then having the caller perform relocation.